packages/space-opera/src/components/camera_settings/reducer.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import {registerStateMutator, State} from '../../redux/space_opera_base.js';
+import {SphericalPositionDeg, Vector3D} from '../../redux/state_types.js';
+
+/*
+ * Register state mutators and get corresponding dispatchers.
+ */
+
+const SET_CAMERA_CONTROLS_ENABLED = 'SET_CAMERA_CONTROLS_ENABLED';
+export const dispatchCameraControlsEnabled = registerStateMutator(
+    SET_CAMERA_CONTROLS_ENABLED, (state: State, enabled?: boolean) => {
+      state.config = {...state.config, cameraControls: !!enabled};
+    });
+
+// Orbit
+const SAVE_CAMERA_ORBIT = 'SAVE_CAMERA_ORBIT';
+export const dispatchSaveCameraOrbit =
+    registerStateMutator(SAVE_CAMERA_ORBIT, (state: State) => {
```

Comment:
I'm going to trust this a copy and paste from above, since github is not helping with the diff.

Reply:
yep, sans the consts